### PR TITLE
[Merged by Bors] - feat(RingTheory/Ideal/Over): define the set of all prime ideals that lie over an ideal

### PR DESCRIPTION
**Pull request by mbkybky**
Define the set of all prime ideals that lie over an ideal. It is mainly for #19141.

This is adapted from the proof in [the case of number fields](https://github.com/jjdishere/neukirch/blob/8fdb75f7ebab8a20eea902058956d7dece648959/AlgebraicNumberTheory/AlgebraicIntegersPart2/HilbertRamificationTheory.lean#L384).

Thanks to @erdOne for reminding me that something similar was done in `flt-regular`. I referred to it to make my code a lot simpler.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Ideal/Over.lean [hunk 1]
```diff
@@ -517,6 +517,29 @@ end CommSemiring
...
+variable {A} {B}
+
+theorem map_ne_bot_of_ne_bot {I : Ideal A} (h : I ≠ ⊥) : map (algebraMap A B) I ≠ ⊥ :=
+  (map_eq_bot_iff_of_injective (NoZeroSMulDivisors.algebraMap_injective A B)).mp.mt h
+
+theorem ne_bot_of_liesOver_of_ne_bot (hp : p ≠ ⊥) (P : Ideal B) [P.LiesOver p] : P ≠ ⊥ := by
+  contrapose! hp
+  apply (over_def P p).trans
+  rw [hp]
+  exact under_bot A B
```
Comment (tb65536) [inline Mathlib/RingTheory/Ideal/Over.lean:539]:
```suggestion
  rw [over_def P p, hp, under_bot]
```

File: Mathlib/RingTheory/Ideal/Over.lean [hunk 2]
```diff
@@ -517,6 +517,29 @@ end CommSemiring
...
+theorem under_bot : under A (⊥ : Ideal B) = ⊥ :=
+  comap_bot_of_injective (algebraMap A B) (NoZeroSMulDivisors.algebraMap_injective A B)
+
+instance bot_liesOver_bot : (⊥ : Ideal B).LiesOver (⊥ : Ideal A) where
+  over := (under_bot A B).symm
+
+variable {A} {B}
+
+theorem map_ne_bot_of_ne_bot {I : Ideal A} (h : I ≠ ⊥) : map (algebraMap A B) I ≠ ⊥ :=
+  (map_eq_bot_iff_of_injective (NoZeroSMulDivisors.algebraMap_injective A B)).mp.mt h
```
Comment (tb65536) [inline Mathlib/RingTheory/Ideal/Over.lean:533]: This theorem should go to an earlier file.

File: Mathlib/RingTheory/Ideal/Over.lean [hunk 3]
```diff
@@ -517,6 +517,29 @@ end CommSemiring
...
 
+variable (A : Type*) [CommRing A] (B : Type*) [Ring B] [Nontrivial B]
+  [Algebra A B] [NoZeroSMulDivisors A B] {p : Ideal A}
+
+@[simp]
+theorem under_bot : under A (⊥ : Ideal B) = ⊥ :=
+  comap_bot_of_injective (algebraMap A B) (NoZeroSMulDivisors.algebraMap_injective A B)
+
+instance bot_liesOver_bot : (⊥ : Ideal B).LiesOver (⊥ : Ideal A) where
+  over := (under_bot A B).symm
```
Comment (tb65536) [inline Mathlib/RingTheory/Ideal/Over.lean:528]: Can you add `under_top` and `top_liesOver_top`?

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**